### PR TITLE
Update Twilio Paste Intellisense to read from published design tokens package, add theme setting

### PR DESCRIPTION
**Pull request by brandongregoryscott**
This PR introduces changes described in #2964 (specifically for items 1 and 2, which should cover the majority of internal users)

Summary of changes:
- Removed hard-coded `PasteToken` objects in favor of using the published objects from `@twilio-paste/design-tokens`
  - A side-effect of this is losing the comment/description that was available of what the token represents, as noted in https://github.com/twilio-labs/paste/discussions/2964#discussioncomment-4777499. If we want to export a different variation of each design token object w/ these comments available, it should be simple to wire up again
- Fixed a bug where hover suggestions for the `boxShadows` prop weren't showing up because it wasn't capitalized properly
- Added a configuration setting named 'Theme' for changing which theme is used to resolve token values (Default, Dark, Evergreen, SendGrid, Twilio and Twilio Dark based on the recently published design-tokens package)

Some screenshots of it in action:
image
image
image
image

---

image
image
image
image

`boxShadow` prop being resolved in the completion item provider 
image

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: apps/vs-code-intellisense/src/utils/theme-to-token-map.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import {PasteToken} from '../types/paste-token';
+import {TokenKey} from '../types/token-key';
+import {TokenMap} from '../types/token-map';
+import {remToPx} from './rem-to-px';
+
+/**
+ * Converts a full theme object into a standard map shape for the extension to serve
+ * completion/hover suggestions for
+ */
+export const themeToTokenMap = (theme: Record<TokenKey, Record<string, string | number>>): TokenMap => {
```
Comment (gkassym) [inline apps/vs-code-intellisense/src/utils/theme-to-token-map.ts:11]: hey, could you write tests for them?

**Comment by gkassym** [comment]
Looks great 👍

File: apps/vs-code-intellisense/src/utils/theme-to-token-map.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  // TODO: Figure out if we can export/import the theme object with descriptions attached to each value
+  // A deep-link to the token on https://paste.twilio.design/tokens/list would also be helpful!
+  comment: undefined,
+});
+
+const toPasteTokenMap = (tokens: Record<string, string | number>): Record<string, PasteToken> => {
+  const keys = Object.keys(tokens);
+  const pasteTokenMap: Record<string, PasteToken> = {};
+  keys.forEach((key) => {
+    pasteTokenMap[key] = toPasteToken(key, tokens[key]);
```
Comment (gkassym) [inline apps/vs-code-intellisense/src/utils/theme-to-token-map.ts:37]: why you didn't use `map`?

File: apps/vs-code-intellisense/src/extension.ts [hunk 3]
```diff
@@ -32,21 +38,12 @@ export function getAttributeName(linePrefix: string) {
...
-    pasteToken === 'borderColors' ||
-    pasteToken === 'colors' ||
-    pasteToken === 'textColors'
-  ) {
-    return true;
-  }
-  return false;
-}
+export const isColor = (tokenKey: TokenKey) =>
+  tokenKey === 'backgroundColors' || tokenKey === 'borderColors' || tokenKey === 'colors' || tokenKey === 'textColors';
```
Comment (gkassym) [inline apps/vs-code-intellisense/src/extension.ts:42]: what do you think to move `isColor` to `utils` folder?

Comment (brandongregoryscott) [reply apps/vs-code-intellisense/src/extension.ts:42]: Yeah, good call! 👍

File: apps/vs-code-intellisense/src/utils/theme-to-token-map.ts (hunk 1 above)
Comment (brandongregoryscott) [reply apps/vs-code-intellisense/src/utils/theme-to-token-map.ts:11]: Sure thing!

**Review by gkassym — APPROVED** [review]
LGTM

File: apps/vs-code-intellisense/src/types/token-category.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  | 'data-visualization'
+  | 'font'
+  | 'font-size'
+  | 'font-weight'
+  | 'line-height'
+  | 'color'
+  | 'sizing'
+  | 'spacing'
+  | 'text-color'
+  | 'z-index';
```
Comment (gkassym) [inline apps/vs-code-intellisense/src/types/token-category.ts:16]: hey, could you explain what is the difference between TokenCategory and TokenType?

File: apps/vs-code-intellisense/src/extension.ts [hunk 5]
```diff
@@ -2,19 +2,20 @@
 // Import the module and reference it with the alias vscode in your code below
 import * as vscode from 'vscode';
 
-import pasteTokens, {pasteTokenAttributes} from './tokens';
-import {PasteToken} from './models/paste-token';
+import {pasteTokenAttributes} from './tokens';
+import {PasteToken} from './types';
+import {getThemeSetting, getThemeTokens, remToPx} from './utils';
+import kebabCase from 'lodash.kebabcase';
```
Comment (gkassym) [inline apps/vs-code-intellisense/src/extension.ts:8]: I imported according to the rules:
1. import libraries `import * as vscode from 'vscode';`
2. import static paths
3. import relative paths `import {PasteToken} from './types';`
Could you keep the same imports?

File: apps/vs-code-intellisense/src/test/suite/utils/get-theme-tokens.test.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import {getThemeTokens} from '../../../utils';
+import assert from 'assert';
```
Comment (gkassym) [inline apps/vs-code-intellisense/src/test/suite/utils/get-theme-tokens.test.ts:2]: import rules

**Review by gkassym — DISMISSED** [review]
I left several comments, but it looks good to me 👍

File: apps/vs-code-intellisense/src/types/token-category.ts (hunk 4 above)
Comment (brandongregoryscott) [reply apps/vs-code-intellisense/src/types/token-category.ts:16]: `type` seems to be the most specific and `category` seems to be the parent grouping. These were pulled from the [`tokens.raw.json`](https://unpkg.com/browse/@twilio-paste/design-tokens@8.5.0/dist/tokens.raw.json) file in the design-tokens package, i.e.

```json
"color-background-user": {
  "type": "color",
  "category": "background-color",
  "value": "#5817bd",
  "comment": "User avatar background color.",
  "originalValue": "{!palette-purple-70}",
  "name": "color-background-user"
}
// ...
"font-size-10": {
  "category": "font-size",
  "type": "font-size",
  "value": "0.625rem",
  "comment": "Constant typography token for font size 10",
  "originalValue": "{!font-size-10}",
  "name": "font-size-10"
}
```

File: apps/vs-code-intellisense/src/extension.ts (hunk 5 above)
Comment (brandongregoryscott) [reply apps/vs-code-intellisense/src/extension.ts]: Should be fixed in each file now!

File: apps/vs-code-intellisense/src/types/token-category.ts (hunk 4 above)
Comment (gkassym) [reply apps/vs-code-intellisense/src/types/token-category.ts:16]: I got it, thanks

**Pull request merged**